### PR TITLE
Add class and method documentation

### PR DESCRIPTION
**Pull request by kevinlind**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt [hunk 1]
```diff
@@ -118,10 +142,6 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
         deleteDeprecatedV5HitDatabase()
     }
 
-    private fun deleteDeprecatedV5HitDatabase() {
```
Comment (kevinlind) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt:121]: Moved after all non-private functions.

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt [hunk 2]
```diff
@@ -375,23 +453,48 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
...
+                Log.debug(
+                    AnalyticsConstants.LOG_TAG,
+                    CLASS_NAME,
+                    "handleAnalyticsRequestIdentityEvent - Failed to parse the visitor identifier to string, ignoring the update visitor identifier request."
+                )
             }
         }
+
         publishAnalyticsId(event)
     }
```
Comment (kevinlind) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt:489]: This is the only code change. It fixes a bug where a null vid value is ignored, however it should clear the existing vid value.

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt [hunk 3]
```diff
@@ -730,26 +896,47 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
...
         if (data.isEmpty()) {
-            Log.debug(AnalyticsConstants.LOG_TAG, CLASS_NAME, "track - event data is null or empty.")
+            Log.debug(AnalyticsConstants.LOG_TAG, CLASS_NAME, "handleTrackRequest - event data is null or empty.")
             return
         }
-        if (data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.TRACK_ACTION) ||
-            data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.TRACK_STATE) ||
-            data.keys.contains(AnalyticsConstants.EventDataKeys.Analytics.CONTEXT_DATA)
-        ) {
+        if (isTrackActionOrTrackStateEvent(data)) {
```
Comment (kevinlind) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt:927]: Call the existing `isTrackActionOrTrackStateEvent()` which performs the same check.

File: code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java [hunk 4]
```diff
@@ -243,11 +245,12 @@ public void fail(AdobeError adobeError) {
...
 
     /**
-     * Retrieves the visitor identifier
+     * Sets the visitor identifier. Clears the visitor identifier when passed null or an empty
+     * string.
      *
      * @param visitorID {@code String} new value for visitor identifier
      */
-    public static void setVisitorIdentifier(@NonNull final String visitorID) {
+    public static void setVisitorIdentifier(@Nullable final String visitorID) {
```
Comment (kevinlind) [inline code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java:253]: Public API change. Previous Android 1.x allowed null values to clear the VID so we want to keep that behavior in 2.x.

Comment (emdobrin) [reply code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java:253]: 👍  make sure Documentation/api-reference is in sync

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsDatabase.kt [hunk 5]
```diff
@@ -19,6 +19,40 @@ import com.adobe.marketing.mobile.services.Log
 import com.adobe.marketing.mobile.services.PersistentHitQueue
 import com.adobe.marketing.mobile.services.ServiceProvider
 
+/**
+ * The Analytics hit database which queues and persists hits before processing.
+ *
+ * Analytics hit reordering
+ * If backDateSessionInfo and offlineTracking is enabled, we should send hit with previous session
+ * information / crash information before sending any hits for current session.
+ * (We get this information from `lifecycle.responseContent` event)
```
Comment (emdobrin) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsDatabase.kt:28]: In general "we" should use impersonal comments, can you reword the comments to not use "we"?

**Review by emdobrin — COMMENTED** [review]
looks super great @kevinlind. I had just couple minor comments

**Review by cacheung — APPROVED** [review]

File: code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java (hunk 4 above)
Comment (kevinlind) [reply code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java:253]: Thanks. I've updated the api-reference documentation.

Comment (emdobrin) [reply code/analytics/src/phone/java/com/adobe/marketing/mobile/Analytics.java:253]: and kotlin String?

File: code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt [hunk 6]
```diff
@@ -437,12 +540,27 @@ internal class AnalyticsExtension(extensionApi: ExtensionApi) : Extension(extens
...
+     */
     private fun isTrackActionOrTrackStateEvent(eventData: Map<String, Any?>): Boolean {
         return eventData.containsKey(AnalyticsConstants.EventDataKeys.Analytics.TRACK_STATE) ||
             eventData.containsKey(AnalyticsConstants.EventDataKeys.Analytics.TRACK_ACTION) ||
             eventData.containsKey(AnalyticsConstants.EventDataKeys.Analytics.CONTEXT_DATA)
     }
 
+    /**
+     * Processes Acquisition track events.
+     * If we are waiting for the acquisition data, then try to append it to a existing hit.
```
Comment (emdobrin) [inline code/analytics/src/main/java/com/adobe/marketing/mobile/analytics/internal/AnalyticsExtension.kt:558]: few more "we"s in this file

**Review by emdobrin — APPROVED** [review]
gorgeous 💯

**Review by addb — APPROVED** [review]

**Pull request merged**